app/scripts/components/common/blocks/scrollytelling/index.tsx

```diff
@@ -134,103 +123,86 @@ function getChapterLayerKey(ch: ScrollyChapter) {
-  // multiple times, even though the data didn't actually change. An example of
-  // this is the `sourceParams` in `MapLayerRasterTimeseries`.
-  // Since the these values only have to be computed once, when the layer loads,
-  // we can use this cache. On every hook run the asyncLayers.map below will
-  // return the cached value if it exists or compute and cache.
-  const resolvedLayersCache = useRef<ResolvedLayer[]>([]);
+    const layer = layers?.find(
+      (l) => l.id === layerId
+    ) as DatasetData | null;
+
+    if (!layer) {
+      throw new Error(
+        `Layer [${layerId}] not found in dataset [${datasetId}]`
+      );
+    }
+
+    return reconcileVizDataset(layer);
+  });
+
+  const [resolvedDatasetsWithStac, setResolvedDatasetsWithStac] = useState<VizDataset[]>([]);
+
+  useReconcileWithStacMetadata(reconciledVizDatasets, setResolvedDatasetsWithStac);
 
   // Each resolved layer will be an object with:
   // layer: The resolved layerData
-  // Component: The component to render the layer
   // runtimeData: The runtime data for the layer
   //
   // The difference between runtimeData and layer is that the layer has the
   // layer definition data, the runtimeData belongs to the application and not
   // the layer. For example the datetime, results from a user action (picking
   // on the calendar or in this case setting it in the MDX).
-  const resolvedLayers = useMemo(
-    () =>
-      asyncLayers.map(({ baseLayer }, index) => {
-        if (baseLayer.status !== S_SUCCEEDED || !baseLayer.data) return null;
-
-        if (resolvedLayersCache.current[index]) {
-          return resolvedLayersCache.current[index];
-        }
-
-        // Some properties defined in the dataset layer config may be functions
-        // that need to be resolved before rendering them. These functions accept
-        // data to return the correct value. Include access to raw data.
-        const datetime = uniqueChapterLayers[index].datetime;
-        const bag = {
-          datetime,
-          dateFns,
-          raw: baseLayer.data
-        };
-        const data = resolveConfigFunctions(baseLayer.data, bag);
-
-        const resolved = {
-          layer: data,
-          Component: getLayerComponent(!!data.timeseries, data.type),
-          runtimeData: {
-            datetime,
-            id: getChapterLayerKey(uniqueChapterLayers[index])
-          }
-        };
-
-        // Need to set it as ResolvedLayer because the "resolveConfigFunctions"
-        // is doing something weird to the tuples and converting something like
-        // "center: [number, number]" to "center: number[]" which fails
-        // validation.
-        resolvedLayersCache.current[index] = resolved as ResolvedLayer;
-
-        return resolved;
-      }),
-    [uniqueChapterLayers, asyncLayers]
-  );
+  const resolvedLayers = resolvedDatasetsWithStac.map((layer, index) => {
+    if (layer.status !== DatasetStatus.SUCCESS) return null;
```

Comment:
❓ : Curious to know if you had a reason to why you got rid of memoization here, if there are no benefits, can probably get rid of it in line 200 as well?

Reply:
Good call, I intended to remove the [caching only](https://github.com/NASA-IMPACT/veda-ui/pull/992#discussion_r1636522808) as I think was obsolete together with the `useMemo`, but looking at the this part again I think it's better to keep the memo part. 

Updated.